src/lib/SoftHSM.cpp

```diff
@@ -89,6 +89,8 @@
 
 // Initialise the one-and-only instance
 
+int objects_deleted = 0;
+
 #ifdef HAVE_CXX11
 
 std::unique_ptr<MutexFactory> MutexFactory::instance(nullptr);
```

Comment:
Don't separate the function from its documentation :)

Reply:
Sorry, I don't get what you mean here.

Reply:
Sorry to be vague. I think the comment `// Initialise the one-and-only instance` belongs to the definition of `std::unique_ptr<MutexFactory> MutexFactory::instance(nullptr);`.